src/NumericalAlgorithms/Interpolation/MultiLinearSpanInterpolation.hpp

```diff
@@ -0,0 +1,387 @@
+    }
+
+    return result;
+  };
+
+  template <size_t... variables_to_interpolate>
+  std::array<double, sizeof...(variables_to_interpolate)> interpolate(
+      Weight<Dimension> const& weights) const {
+    static_assert(sizeof...(variables_to_interpolate) <= NumberOfVariables,
+                  "You are trying to interpolate more variables than this "
+                  "container holds.");
+
+    return std::array<double, sizeof...(variables_to_interpolate)>{
+        interpolate(weights, variables_to_interpolate)...};
+  }
+
+  template <size_t NumberOfvariables_to_interpolate, typename... T,
+            Requires<(std::is_floating_point_v<typename std::remove_cv_t<T>> and
+                      ...)> = nullptr>
+  std::array<double, NumberOfvariables_to_interpolate> interpolate(
+      std::array<size_t, NumberOfvariables_to_interpolate>&
+          variables_to_interpolate,
+      T const&... target_points) const {
+    static_assert(
+        sizeof...(T) == Dimension,
+        "You need to provide the correct number of interpolation points");
+    static_assert(NumberOfvariables_to_interpolate <= NumberOfVariables,
+                  "You are trying to interpolate more variables than this "
+                  "container holds.");
+
+    auto weights = getWeights(target_points...);
+
+    std::array<double, NumberOfvariables_to_interpolate> interpolatedValues;
+
+    for (size_t nn = 0; nn < NumberOfvariables_to_interpolate; ++nn) {
+      interpolatedValues[nn] =
+          interpolate(weights, variables_to_interpolate[nn]);
+    }
+
+    return interpolatedValues;
+  }
+
+  template <size_t NumberOfvariables_to_interpolate, size_t... I>
+  std::array<double, NumberOfvariables_to_interpolate> interpolate(
+      std::array<size_t, NumberOfvariables_to_interpolate>&
+          variables_to_interpolate,
+      std::array<double, Dimension>& target_points,
+      std::index_sequence<I...>) const {
+    return interpolate(variables_to_interpolate, target_points[I]...);
+  }
+
+  template <size_t NumberOfvariables_to_interpolate>
+  std::array<double, NumberOfvariables_to_interpolate> interpolate(
+      std::array<size_t, NumberOfvariables_to_interpolate>&
+          variables_to_interpolate,
+      std::array<double, Dimension>& target_points) const {
+    return interpolate(variables_to_interpolate, target_points,
+                       std::make_index_sequence<Dimension>{});
+  }
+
+  MultiLinearSpanInterpolation(
+      std::array<gsl::span<double const>, Dimension> x_,
+      gsl::span<double const> y_, Index<Dimension> _number_of_points_);
+
+  double lower_bound(const size_t which_dimension) const {
+    return _x[which_dimension][0];
+  };
+
+  double upper_bound(const size_t which_dimension) const {
+    return _x[which_dimension][_number_of_points[which_dimension] - 1];
+  };
+
+ private:
+  /// Allow extrapolation above table bounds.
+  std::array<bool, Dimension> _allow_extrapolation_below_data;
```

Comment:
we use `_` after the variable name rather than before for member variables (apply to all below please)